client/notifications/index.jsx

```diff
@@ -206,80 +259,101 @@ var Notifications = React.createClass( {
-			this.queuedMessage = message;
-		}
-	},
 
-	postServiceWorkerMessage: function( message ) {
-		if ( 'serviceWorker' in window.navigator ) {
-			window.navigator.serviceWorker.ready.then( ( serviceWorkerRegistration ) => {
-				if ( 'active' in serviceWorkerRegistration ) {
-					serviceWorkerRegistration.active.postMessage( message );
-				}
-			} );
-		}
-	},
-
-	render: function() {
-		var userData = user.get(),
-			localeSlug = userData.localeSlug || config( 'i18n_default_locale_slug' ),
-			widgetURL = widgetDomain,
-			frameClasses = [ 'wide' ],
-			panelClasses = [ 'wide' ],
-			now = new Date();
-
-		if ( config.isEnabled( 'notifications2beta' ) ) {
-			widgetURL = widgetURL + '/notificationsbeta/';
-		} else {
-			widgetURL = widgetURL + '/notifications/';
+				return;
 		}
-		if ( user.isRTL() ) {
-			widgetURL += 'rtl.html';
+	};
+
+	/**
+	 * Sends a message to the notifications client
+	 *
+	 * @param {!Object} message data to send
+	 * @param {!String} message.action name of action for notes app to dispatch
+	 * @returns {*} please ignore return value
+	 */
+	postMessage = message => {
+		// save only the latest message to send when iframe is loaded
+		if ( ! ( this.notesFrame && this.state.iframeLoaded ) ) {
+			return this.queuedMessage = message;
 		}
-		widgetURL += '?locale=' + localeSlug;
 
-		// cache buster
-		widgetURL += '&' + now.getFullYear() + ( now.getMonth() + 1 ) + now.getDate() + ( now.getHours() + 10 );
+		const data = JSON.stringify( {
+			...message,
+			type: 'notesIframeMessage',
+		} );
 
-		if ( this.state.widescreen && this.props.visible ) {
-			frameClasses.push( 'widescreen' );
-		}
+		this.notesFrame.contentWindow.postMessage( data, widgetDomain );
+	};
 
-		if ( this.props.visible ) {
-			panelClasses.push( 'wpnt-open' );
-		} else {
-			panelClasses.push( 'wpnt-closed' );
+	postServiceWorkerMessage = message => {
+		if ( ! ( 'serviceWorker' in window.navigator ) ) {
+			return;
 		}
 
+		window.navigator.serviceWorker.ready.then(
+			registration => ( 'active' in registration ) && registration.active.postMessage( message )
+		);
+	};
+
+	storeNotesFrame = ref => this.notesFrame = ref;
+
+	render() {
 		if ( ! this.props.visible && ! this.state.loaded ) {
+			// @TODO we need a good loading message
 			return <div />;
 		}
 
+		const localeSlug = get( user.get(), 'localeSlug', config( 'i18n_default_locale_slug' ) );
+		const now = new Date();
+
+		/** * @type {string} holds the URL for the notifications client for the iframe */
+		const widgetUrl = [
+			widgetDomain,
+			'/notifications',
+			// we deploy to beta before pushing to production
+			config.isEnabled( 'notifications2beta' ) && 'beta',
+			'/',
+			// default app is at index.html
+			// auto-rtl is used to generate
+			// the flipped copy
+			user.isRTL() && 'rtl.html',
+			`?locale=${ localeSlug }`,
+			// bust cache every hour
+			// @TODO replace this with a strong process that doesn't make users download so much data
```

Comment:
In the future I would recommend using the same `git describe` output from the notifications-client repo